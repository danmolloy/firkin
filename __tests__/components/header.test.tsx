import Header, {  menuItems } from "@/components/header"
import { act, fireEvent, render, screen } from "@testing-library/react"
import { Link, animateScroll } from "react-scroll";
import "@testing-library/jest-dom"

jest.mock("react-scroll", () => ({
  animateScroll: {
    scrollToTop: jest.fn()
  },
  Link: (props: any) => <div></div>
}))


describe("<Header />", () => {
  beforeEach(() => {
    const mockIntersectionObserver = jest.fn();
  mockIntersectionObserver.mockReturnValue({
    observe: () => null,
    unobserve: () => null,
    disconnect: () => null
  });
  window.IntersectionObserver = mockIntersectionObserver;

    render(<Header  />)
  })

  it("header-section is in the document", () => {
    const header = screen.getByTestId("header-section")
    expect(header).toBeInTheDocument()
  })
  it("band name is in the document, which scrolls to top on click", () => {
    const bandName = screen.getByText("The Big Firkin Band")
    expect(bandName).toBeInTheDocument()
    act(() => {
      fireEvent.click(bandName)
    })
    expect(animateScroll.scrollToTop).toHaveBeenCalledWith({
      smooth: true, 
      duration: 500
    })
  }) 
  it("<DonateButton /> is in the document", () => {
    const donateBtn = screen.getByTestId("donate-btn")
    expect(donateBtn).toBeInTheDocument()
  })
  it("matches snapshot", () => {
    const header = screen.getByTestId("header-section")
    expect(header).toMatchSnapshot()

  })
})