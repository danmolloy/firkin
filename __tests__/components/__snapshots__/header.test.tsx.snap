// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<div
  class="text-black  fixed w-full font-display border-b border-gray-600 h-14 flex flex-row justify-between items-center px-4 backdrop-blur z-20"
  data-testid="header-section"
>
  <button
    class="text-black font-display"
  >
    The Big Firkin Band
  </button>
  <form
    action="/donate/api"
    data-testid="donate-btn"
    method="POST"
  >
    <button
      class="hover:text-yellow-400 rounded font-bold text-lg font-text"
      role="link"
      type="submit"
    >
      Donate
    </button>
  </form>
</div>
`;
